029-IoTEdge/Student/Challenge-07.md

```diff
@@ -0,0 +1,116 @@
+
+## Description
+
+The [Device Streams documentation](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-device-streams-overview) has a complete description of how Device Streams work. In a simplified way, they allow clients to connect to devices via the bi-directional connection established from the device to an Azure IoT Hub. This connection is outbound (opened by the device), and thus even if a firewall blocks inbound connections, the remote connections will still be possible. You will need to have something running in the device (which we'll call "device-local proxy") to handle these connections.
+
+For this challenge, you'll be using Device Streams to set up an `SSH` connection to the simulated IoT Device you've been using, and then locking down firewall to confirm that a connection is still possible.
+
+Steps:
+
+1. Run the steps of the quickstart [Enable SSH and RDP over an IoT Hub device stream by using a C# proxy application (preview)](https://docs.microsoft.com/en-us/azure/iot-hub/quickstart-device-streams-proxy-csharp#ssh-to-a-device-via-device-streams), in particular the section "SSH to a device via device streams". Make sure you test them and that you are able to run commands on the device via the Device Streams tunnel.
+1. Block inbound SSH traffic (port 22) in your IoT Device and confirm the device is still reachable using what you created for step 1, but not if you try to reach it directly via its IP.
+1. Check the Advanced Challenge if you want to have the device-local proxy running as a Linux system service.
+
+After completing the steps above, you'll have learned about Device Streams and how they allow you to leverage Azure IoT Hub in its security and networking capabilities to remotely connect into devices without requiring third party software. The Advanced Challenge guides you through getting the device-local proxy running as a service so that it's always available.
+
+## Success Criteria
+
+1. You are able to successfully connect via `SSH` over IoT Hub to your simulated IoT Device
+1. You have closed down SSH on the firewall and the connection was not impacted
+
+## Learning Resources
+
+- [IoT Hub Device Streams (Microsoft Docs)](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-device-streams-overview)
+- [Introduction to Azure IoT learning path (Microsoft Learn)](https://docs.microsoft.com/en-us/learn/paths/introduction-to-azure-iot/)
+
+## Tips
+
+- When running the quickstart, before building the code, if you have a recent version of .Net Core, make sure the .csproj file for the device-local proxy (`~/devicestreams/iot-hub/Quickstarts/device-streams-proxy/device`) has `TargetFramework` as `netcoreapp3.1`.
+- If you're installing .Net Core for the first time, you'll need to do a reboot.
+
+## Advanced Challenges
```

Comment:
Same comment here as before, make it consistent..... either put "(optional)" on all of them or none of them.